frontend/model/contracts/shared/types.js

```diff
@@ -64,18 +64,21 @@ export const messageType: any = objectMaybeOf({
     type: unionOf(...Object.values(MESSAGE_NOTIFICATIONS).map(v => literalOf(v))),
     params: mapOf(string, string) // { username }
   }),
-  attachments: arrayOf(objectOf({
-    name: string,
-    mimeType: string,
-    dimension: optional(objectOf({
-      width: number,
-      height: number
-    })),
-    downloadData: objectOf({
-      manifestCid: string,
-      downloadParams: optional(object)
-    })
-  })),
+  attachments: optional(
+    arrayOf(objectOf({
+      name: string,
+      mimeType: string,
+      size: optional(numberRange(1, Number.MAX_SAFE_INTEGER)),
```

Comment:
Can you check if we always provide this size info now? If so, this should probably no longer be optional.